lio-http-server/src/LIO/HTTP/Server/Frankie.hs

```diff
@@ -248,31 +247,26 @@ instance Ord PathSegment where
 port :: Port -> FrankieConfigMode s m ()
 port p = do
   cfg <- getModeConfig
-  when (isJust $ cfgPort cfg) $ cfgFail "port already set"
-  setModeConfig $ cfg {cfgPort = Just p }
+  setModeConfig $ cfg {cfgPort = cfgPort cfg <> pure p }
 
 -- | Set the app host preference.
 host :: HostPreference -> FrankieConfigMode s m ()
 host pref = do
   cfg <- getModeConfig
-  -- XXX can we use liquid types instead?
-  when (isJust $ cfgHostPref cfg) $ cfgFail "host already set"
```

Comment:
A stack trace would for sure be useful, but ignoring the error is absolutely worse than giving someone a runtime error.